src/test/clojure/jarl/builtins/bits_test.clj

```diff
@@ -1,97 +1,62 @@
 (ns jarl.builtins.bits_test
-  (:require [clojure.test :refer [deftest is testing]]
-            [jarl.builtins.bits :refer [builtin-bits-or builtin-bits-and builtin-bits-negate
-                                        builtin-bits-xor builtin-bits-lsh builtin-bits-rsh]])
-  (:import (se.fylling.jarl UndefinedException)))
+  (:require [clojure.test :refer [deftest]]
+            [test.utils :refer [testing-builtin]])
+  (:import (se.fylling.jarl TypeException)))
 
 (deftest builtin-bits-or-test
-  (testing "bits.or"
-    (is (= (builtin-bits-or 13 10) 15))
-    (is (= (builtin-bits-or 10 -10) -2))
-    (is (= (builtin-bits-or -10 -10) -10)))
-  (testing "non-integers"
-    (is (thrown-with-msg? UndefinedException
-                          #"eval_type_error: bits.or: operand 1 must be integer number but got floating-point number"
-                          (builtin-bits-or 10.1 1)))
-    (is (thrown-with-msg? UndefinedException
-                          #"eval_type_error: bits.or: operand 2 must be integer number but got floating-point number"
-                          (builtin-bits-or 2 2.22)))))
+  (testing-builtin "bits.or"
+    [13 10]   15
+    [10 -10]  -2
+    [-10 -10] -10
+    [10.1 1]  [TypeException "operand 1 must be integer number but got floating-point number"]
+    [2 2.22]  [TypeException "operand 2 must be integer number but got floating-point number"]))
 
 (deftest builtin-bits-and-test
-  (testing "bits.and"
-    (is (= (builtin-bits-and 13 10) 8))
-    (is (= (builtin-bits-and 10 -10) 2))
-    (is (= (builtin-bits-and 100 5) 4)))
-  (testing "non-integers"
-    (is (thrown-with-msg? UndefinedException
-                          #"eval_type_error: bits.and: operand 1 must be integer number but got floating-point number"
-                          (builtin-bits-and 10.1 1)))
-    (is (thrown-with-msg? UndefinedException
-                          #"eval_type_error: bits.and: operand 2 must be integer number but got floating-point number"
-                          (builtin-bits-and 2 2.22)))))
+  (testing-builtin "bits.and"
+    [13 10]  8
+    [10 -10] 2
+    [100 5]  4
+    [10.1 1] [TypeException "operand 1 must be integer number but got floating-point number"]
+    [2 2.22] [TypeException "operand 2 must be integer number but got floating-point number"]))
 
 (deftest builtin-bits-negate-test
```

Comment:
I was starting to think that all tests that exercise an exception fails, but this one doesn't. All other tests in this file does, however.